src/AppBundle/Twig/AppExtension.php

```diff
@@ -28,9 +29,12 @@ class AppExtension extends \Twig_Extension
 {
     private $parser;
 
-    public function __construct(Markdown $parser)
+    private $purifier;
+
+    public function __construct(Markdown $parser, HTMLPurifierExtension $purifier)
```

Comment:
injecting a Twig extension into another one does not make sense. you should inject HtmlPurifier itself